src/optimizer/pushdown/pushdown_get.cpp

```diff
@@ -1,27 +1,119 @@
+	switch (filter->type) {
+	case ExpressionType::CONJUNCTION_OR: {
+		auto &or_exp = (BoundConjunctionExpression &)*filter;
+		for (auto &child : or_exp.children) {
+			get_equality_constant_filters(child.get(), cons_eq_filters);
+		}
+		break;
+	}
+	case ExpressionType::CONJUNCTION_AND: {
+		auto &and_exp = (BoundConjunctionExpression &)*filter;
+		for (auto &child : and_exp.children) {
+			get_equality_constant_filters(child.get(), cons_eq_filters);
+		}
+		break;
+	}
+	case ExpressionType::COMPARE_EQUAL: {
+		auto &comp_exp = (BoundComparisonExpression &)*filter;
+		if ((comp_exp.left->expression_class == ExpressionClass::BOUND_COLUMN_REF &&
+		     comp_exp.right->expression_class == ExpressionClass::BOUND_CONSTANT)) {
+			auto &column_ref = (BoundColumnRefExpression &)*comp_exp.left;
+			cons_eq_filters[column_ref.binding.column_index] = comp_exp.right->Copy();
+		}
+		break;
+	}
+	default:
+		break;
+	}
+}
+
+void replace_column_constants(Expression *filter, unordered_map<idx_t, unique_ptr<Expression>> &cons_eq_filters) {
+	switch (filter->type) {
+	case ExpressionType::CONJUNCTION_OR: {
+		auto &or_exp = (BoundConjunctionExpression &)*filter;
+		for (auto &child : or_exp.children) {
+			get_equality_constant_filters(child.get(), cons_eq_filters);
+		}
+		break;
+	}
+	case ExpressionType::CONJUNCTION_AND: {
+		auto &and_exp = (BoundConjunctionExpression &)*filter;
+		for (auto &child : and_exp.children) {
+			get_equality_constant_filters(child.get(), cons_eq_filters);
+		}
+		break;
+	}
+	case ExpressionType::COMPARE_EQUAL:
+	case ExpressionType::COMPARE_LESSTHAN:
+	case ExpressionType::COMPARE_LESSTHANOREQUALTO:
+	case ExpressionType::COMPARE_GREATERTHANOREQUALTO:
+	case ExpressionType::COMPARE_GREATERTHAN: {
+		auto &comp_exp = (BoundComparisonExpression &)*filter;
+		if (comp_exp.left->expression_class == ExpressionClass::BOUND_COLUMN_REF) {
+			if (comp_exp.right->expression_class == ExpressionClass::BOUND_CAST) {
+				auto &cast = (BoundCastExpression &)*comp_exp.right;
+				if (cast.child->expression_class == ExpressionClass::BOUND_COLUMN_REF) {
+					auto &column_ref = (BoundColumnRefExpression &)*cast.child;
+					auto it = cons_eq_filters.find(column_ref.binding.column_index);
+					if (it != cons_eq_filters.end()) {
+						//! We can replace it.
+						comp_exp.right = it->second->Copy();
+					}
+				}
+			} else if (comp_exp.right->expression_class == ExpressionClass::BOUND_COLUMN_REF) {
+				auto &column_ref = (BoundColumnRefExpression &)*comp_exp.right;
+				auto it = cons_eq_filters.find(column_ref.binding.column_index);
+				if (it != cons_eq_filters.end()) {
+					//! We can replace it.
+					comp_exp.right = it->second->Copy();
+				}
+			}
+		}
+		break;
+	}
+	default:
+		break;
+	}
+}
 
+void rewrite_transitive_filters(vector<unique_ptr<Filter>> &filters) {
+	unordered_map<idx_t, unique_ptr<Expression>> cons_eq_filters;
```

Comment:
Adding transitive filters should already be handled by the FilterCombiner using equivalence sets. Could you try using the FilterCombiner here to implement this functionality? If that doesn't work perhaps check what is missing there?